### PR TITLE
Support for procedure method override

### PR DESCRIPTION
**Pull request by jlalmes**
Closes https://github.com/trpc/trpc/issues/1939.

## Support for procedure method override

By passing an optional `method?: 'GET' | 'POST'` property into `TRPCRequestOptions`, you can override the HTTP transport method used by queries/mutations. This is great for circumventing URL length limits when sending large input payloads to `query` procedures (there are use-cases for sending a `mutation` via GET too).

### Implementation
If `method` override is present the original `type` is added as a query parameter (i.e. `?type=query`). The `@trpc/server` will then resolve the `input` as the `POST` body but proceed to handle the request as a `query`. 

### Example
Query as POST: `client.query('getData', { bigPayload: [...] }, { method: 'POST' })`

### Additional
`httpLink` & `httpBatchLink` will now throw an error when sent a `subscription` operation.

### CODE REVIEW

File: packages/client/src/internals/httpRequest.ts [hunk 1]
```diff
@@ -15,17 +14,17 @@ function arrayToDict(array: unknown[]) {
 export function httpRequest<TResponseShape = TRPCResponse>(
   props: {
     runtime: LinkRuntimeOptions;
-    type: ProcedureType;
+    type: 'query' | 'mutation';
+    method: 'query' | 'mutation';
```
Comment (mmkal) [inline packages/client/src/internals/httpRequest.ts:18]: I think `'get' | 'post'` Are more intuitive here don’t you think?

Comment (jlalmes) [reply packages/client/src/internals/httpRequest.ts:18]: Option A
```suggestion
    method: 'GET' | 'POST';
```
Option B
```suggestion
    typeOverride: 'query' | 'mutation';
```

Thoughts?

Comment (mmkal) [reply packages/client/src/internals/httpRequest.ts:18]: I prefer A, it more directly tells the end user what the property will actually do. Aside from changing get to post it doesn’t somehow turn a query into a mutation. It’s purely an http thing right?

**Comment by jlalmes** [comment]
Just updated description and added tests - ready for review!

**Review by KATT — COMMENTED** [review]
This could be used for DoS-attacks, could we make it opt-out on the server the same wait we can opt-out of batching?

**Review by sachinraja — APPROVED** [review]
Looks good - I agree with Alex's comment and just left a few code quality comments. I think we should couple this with good docs to prevent unnecessary usage and hopefully the new transformer API can also help here. I still think it's useful for production apps despite opening up DoS attacks and some providers allow you to cache POST requests.

**Comment by jlalmes** [comment]
* Normalized method in `TRPCClient.ts`.
* Deduped HTTP subscription error message.
* Added opt-out for server via `methodOverride.enabled: false`.
* Fixed tests & added test for opt-out.

File: packages/client/src/internals/executeChain.ts [hunk 2]
```diff
@@ -62,6 +63,8 @@ export function executeChain<
       },
     });
   }
+
   walk({ index: 0, op: opts.op, stack: [] });
+
```
Comment (KATT) [inline packages/client/src/internals/executeChain.ts:68]:
```suggestion
```

**Comment by KATT** [comment]
Closing because of the many conflicts it has, feel free to pick it up though

**Pull request closed without merging**